### ISSUE TITLE
win_share - an error occurred when attempting to create share Ansible 2.3 RC1

### ISSUE THREAD

**Issue by jborean93** [ansible/ansible#22730]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
win_share

##### ANSIBLE VERSION
```
ansible 2.3.0.0 (stable-2.3 923c9ef17c) last updated 2017/03/17 09:12:40 (GMT +1000)
  config file = /apps/dev/playbook/jordan/mr-ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.12 (default, Sep 23 2016, 14:23:49) [GCC 4.4.7 20120313 (Red Hat 4.4.7-17)]
```

##### CONFIGURATION
Standard

##### OS / ENVIRONMENT
RHEL 6.7 - Windows 2012 R2

##### SUMMARY
When trying to create a Windows share you get a very generic error message `an error occurred when attempting to create share SHARENAME`. When modifying win_share to get a more helpful error we get the full error as to what happened

```
 Cannot process argument transformation on parameter 'CachingMode'. Cannot convert value "" to type "Microsoft.PowerShell.Cmdletization.GeneratedTypes.SmbShare.CachingMode". Error: "The identifier name  cannot be processed because it is either too similar or identical to the following enumerator names: None, Manual, Documents, Programs, BranchCache, Unknown. Use a more specific identifier name.
```

##### STEPS TO REPRODUCE
Need to find out more reasons why this is happening but from an initial investigation the addition of the caching mode changes https://github.com/ansible/ansible/pull/21103 seems to have caused this issue. Will look further later this weekend as to how to fix it and implement PR for it.

**Comment by daBONDi** [comment]
How is the Playbook looks what you put into the Param "CachingMode" ?

Posibility to post the play somewhere?

**Comment by daBONDi** [comment]
someone add this to milestone 2.3.0, its critical

**Comment by daBONDi** [comment]
Added PR for
devel: #22753 
stable-2.3: #22755

to fix this issue

**Comment by nitzmahone** [comment]
Merged #22753 merged and cherry-picked to stable-2.3 for 2.3.0RC2
